lua/autorun/client/cl_cfc_restart_display.lua

```diff
@@ -0,0 +1,71 @@
+local finalTime
+local keyFrame = 0
+local isOpen = false
+local flashState = false
+local CFCRestart = {}
+surface.CreateFont( "CFC_CALIBRI_20PX_DEFAULT", {
+    font = "Calibri",
+    extended = false,
+    size = 40,
+    weight = 500
+})
+
+local function timeLeft()
+    return finalTime - SysTime()
+end
+
+local function restartUI()
+    timer.Create( "CFC_RESTART_FLASHTIMER", 1, 0, function()
+        flashState = not flashState
+    end )
+
+    isOpen = true
+    hook.Remove( "HUDPaint", "CFCDrawRestartAlert" )
+    local delay = net.ReadInt( 16 )
+    local curTime = SysTime()
+    finalTime = curTime + delay
+    hook.Add( "HUDPaint", "CFCDrawRestartAlert", function()
```

Comment:
Is it possible to have this hook defined elsewhere? You could pull the function out into something named like `restartHUDPaint()` then just do `hook.Add( "HUDPaint", "CFCDrawRestartAlert", restartHUDPaint )` to make this cleaner.
I'm not entirely sold on the name of that function, so I'll leave it up to you to think of something more fitting.